contracts/utils/helpers.cairo

```diff
@@ -144,12 +144,14 @@ func assert_game_exists{
         pedersen_ptr : HashBuiltin*,
         range_check_ptr
     }(
-        game : felt
+        game : felt,
+        generation : felt
     ):
     with_attr error_message(
         "Game {game} does not exist"
     ):
-        assert_not_zero(game)
+        let (current_gen) = current_generation.read(game)
+        assert_le_felt(generation, current_gen)
```

Comment:
If game doesn't exist you'll get `current_gen = 0`, right? So you'll say that all non exsiting games have a generation 0.

Reply:
Right, so I should also check if the current_gen is not zero here. Added in newest commit.